tests/python/core/test_multi_function_flow.py

```diff
@@ -0,0 +1,249 @@
+
+    if not isinstance(model_or_mod, tvm.IRModule):
+        record = model_or_mod._internal(*args)
+        args = _get_func_inputs(record, args, {}, get_handle=False)
+        mod = record.mod
+    else:
+        mod = model_or_mod
+
+    # Use CPU to avoid workspace memory.
+    device = "cpu"
+    InitPool(Device(device), "page_unit_pool")
+    pass_config = {"raf.memory_schedule": True, "raf.memory_budget": int(13e9)}
+    if use_multi_func:
+        pass_config["raf.use_multi_func"] = True
+
+    with tvm.transform.PassContext(
+        opt_level=3, config=pass_config, disabled_pass=["FuseTVM", "FuseDialect"]
+    ):
+        res = VMExecutor(mod, device).make_executor()(*args)
+
+    return res.numpy()
+
+
+# Simple convnet to test the multi-function flow
+def test_simple_convnet():
+
+    device = "cpu"
+    ishape = (16, 16, 64, 64)
+    pooled_ishape = (16, 16, 32, 32)
+    dense_ishape = (16, 16)
+    wgtshape = (16, 16, 3, 3)
+    dense_wgt_shape = (1, 16)
+
+    # Get operators
+    conv2d_op = raf._ffi.op.GetOp("raf.op.conv2d")
+    conv2d_call = lambda x, w: relay.Call(
+        conv2d_op,
+        [
+            x,
+            w,
+            raf.ir.const([1]),
+            raf.ir.const([1]),
+            raf.ir.const([1]),
+            raf.ir.const(1),
+            raf.ir.const("NCHW"),
+            raf.ir.const("OIHW"),
+            raf.ir.const("NCHW"),
+        ],
+    )
+    max_pool2d_op = raf._ffi.op.GetOp("raf.op.max_pool2d")
+    max_pool2d_call = lambda x, winsize: relay.Call(
+        max_pool2d_op,
+        [
+            x,
+            raf.ir.const(winsize),  # kernel size
+            raf.ir.const(winsize),  # stride
+            raf.ir.const([0]),  # padding
+            raf.ir.const([1]),  # dilation
+            raf.ir.const(False),
+            raf.ir.const(True),
+            raf.ir.const("NCHW"),
+        ],
+    )
+
+    def get_mod_multi_func():
+        """
+        Create a simple convnet model with two types of layers. These two
+        layers have different input shapes. This test is without autodiff.
+        More comprehensive testing with autodiff is postponed.
+
+        fn(input, wgt0, wgt1, wgt2, wgt3, wgt4) {
+
+            // Layer 0
+            let foo0 = layer0(inp0, wgt0) {
+                let conv_out0 = conv2d(inp0, wgt0)
+                let relu_out0 = relu(conv_out0)
+                relu_out0
+            }
+
+            // Layer 1
+            let foo1 = layer1(inp1, wgt1) {
+                let conv_out1 = conv2d(inp1, wgt1)
+                let relu_out1 = relu(conv_out1)
+                relu_out1
+            }
+
+            // First two layers
+            let a0 = foo0(input, wgt0)
+            let a1 = foo0(a0, wgt1)
+
+            // Go through a pooling layer
+            let a1_pooled = maxpool_2d(a1, [2, 2])
+
+            // Next two layers
+            let a2 = foo1(a1_pooled, wgt2)
+            let a3 = foo2(a2, wgt3)
```

Comment:
I think here should be like this if you are to call `foo1` or actually layer1 twice.
```
let a3 = foo1(a2, wgt3) // or let a3 = layer1(a2, wgt3)
```

Reply:
Good catch. This is a typo and`foo2` should be replaced by `foo1`.